lib/ansible/modules/net_tools/nmcli.py

```diff
@@ -556,7 +556,7 @@ def __init__(self, module):
         self.type = module.params['type']
         self.ip4 = module.params['ip4']
         self.gw4 = module.params['gw4']
-        self.dns4 = ' '.join(module.params['dns4'])
+        self.dns4 = ' '.join(module.params['dns4']) if module.params.get('dns4') else None
```

Comment:
Looks good to me. Maybe add a test?

Reply:
There is no existing test for nmcli and I am not aware of all the possible scenarios of nmcli, so may not be the right person to add new tests.

Reply:
No need to add other scenarios, it's fine to add a test for just this one use case, but agreed that `nmcli` might be tricky to test.